Freebsdcpud

rminnich opened the pull request.

rminnich committed:
cpud for freebsd

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

binjip978 commented on cmds/cpud/init_freebsd.go:
```diff
@@ -0,0 +1,81 @@
+	}()
+
+	runtime.UnlockOSThread()
+	return nil
```
I don't see how this function can return an error. But is this more or less duplicate, I think it's fine.

binjip978 commented on cmds/cpud/main_freebsd.go:
```diff
@@ -0,0 +1,131 @@
+		}
+		// If we are here, no matter what they may set, *remote must be true.
+		// sadly, cpud -d -remote=true -remote=false ... works.
+		*remote = true
```
no sure if I get this, but for me line 69 defines remote value, maybe we should not parse it at all? I mean remove `remote = flag.Bool("remote", false, "indicates we are the remote side of the cpu session")`

rminnich replied:
If we can ever run as init on freebsd, this will matter, so I'd rather leave it in. It follows the pattern of the linux side.

rminnich committed:
ignore .pub files

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

rminnich committed:
checkpoint -- deal with freebsd silliness.

freebsd seems to have issues being told what to use for the mount port.
Most annoying.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

rminnich committed:
checkpoint -- it actually kind of works but

freebsd lacking private name spaces and bind mounts
makes this a bit of a pain.

The best way forward may be a jail.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>
